Mathlib/Data/Finsupp/Lex.lean

```diff
@@ -84,6 +86,11 @@ instance Lex.linearOrder [LinearOrder N] : LinearOrder (Lex (α →₀ N)) where
   le := (· ≤ ·)
   __ := LinearOrder.lift' (toLex ∘ toDFinsupp ∘ ofLex) finsuppEquivDFinsupp.injective
 
+theorem lex_le_iff_of_unique [Unique α] [PartialOrder N] {x y : Lex (α →₀ N)} :
+    x ≤ y ↔ x default ≤ y default := by
+  rw [le_iff_lt_or_eq, le_iff_lt_or_eq, lex_lt_iff_of_unique, ← ofLex.apply_eq_iff_eq,
+    DFunLike.ext_iff, Unique.forall_iff]
```

Comment:
```suggestion
theorem lex_le_iff_of_unique [Unique α] [PartialOrder N] {x y : Lex (α →₀ N)} :
    x ≤ y ↔ x default ≤ y default :=
  Pi.lex_le_iff_of_unique
```
after the [generalisation below](https://github.com/leanprover-community/mathlib4/pull/30973/files#r2550174281)